From e9a9e29110fc81061bc99bbb884f1ae251c9fc07 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 2 May 2014 14:36:54 -0400
Subject: [misc] mei: print correct device state during unexpected reset

Message-id: <1399041470-21330-22-git-send-email-prarit@redhat.com>
Patchwork-id: 79309
O-Subject: [RHEL7.1 PATCH BZ 1088953 21/77] mei: print correct device state during unexpected reset
Bugzilla: 1088953
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1088953

commit f931f4f3f0e2be1a6c5278ab9488652f98ba2f9c
Author: Alexander Usyskin <alexander.usyskin@intel.com>
Date:   Mon Oct 21 22:05:43 2013 +0300

    mei: print correct device state during unexpected reset

    Move the unexpected state print to the beginning of mei_reset,
    thus printing right state.

    Signed-off-by: Alexander Usyskin <alexander.usyskin@intel.com>
    Signed-off-by: Tomas Winkler <tomas.winkler@intel.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Cc: Dean Nelson <dnelson@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/misc/mei/init.c b/drivers/misc/mei/init.c
index 64fd949..5f60430 100644
--- a/drivers/misc/mei/init.c
+++ b/drivers/misc/mei/init.c
@@ -147,6 +147,10 @@ void mei_reset(struct mei_device *dev, int interrupts_enabled)
    dev->dev_state != MEI_DEV_POWER_DOWN &&
    dev->dev_state != MEI_DEV_POWER_UP);
 
+ if (unexpected)
+  dev_warn(&dev->pdev->dev, "unexpected reset: dev_state = %s\n",
+    mei_dev_state_str(dev->dev_state));
+
  ret = mei_hw_reset(dev, interrupts_enabled);
  if (ret) {
   dev_err(&dev->pdev->dev, "hw reset failed disabling the device\n");
@@ -185,10 +189,6 @@ void mei_reset(struct mei_device *dev, int interrupts_enabled)
  dev->rd_msg_hdr = 0;
  dev->wd_pending = false;
 
- if (unexpected)
-  dev_warn(&dev->pdev->dev, "unexpected reset: dev_state = %s\n",
-    mei_dev_state_str(dev->dev_state));
-
  if (!interrupts_enabled) {
   dev_dbg(&dev->pdev->dev, "intr not enabled end of reset\n");
   return;
-- 
1.7.1